app/models/os_major_version.rb

```diff
@@ -45,14 +49,25 @@ def self.aggregated_cast(column)
                         inverse_of: false, dependent: :restrict_with_exception
 
   has_many :profiles, lambda {
+    # List the profiles in use for the current user
+    profiles_in_use = User.current.account.profiles.joins(PROFILE_BM_JOIN.join_sources)
+                          .select(arel_table[:ref_id], OS_MAJOR_VERSION, PROFILE_IN_USE)
+
+    # Canonical profiles joined with profiles in use and grouped by supported benchmark versions
     supported_profiles = canonical.where(upstream: false)
-                                  .joins(:benchmark)
-                                  .select(PROFILE_LAST_ID, PROFILE_BM_VERSIONS, OS_MAJOR_VERSION)
-                                  .group(:ref_id, 'os_major_version')
+                                  .joins(PROFILE_BM_JOIN.join_sources)
+                                  .joins("LEFT OUTER JOIN (#{profiles_in_use.to_sql}) sq ON
+                                    sq.ref_id = profiles.ref_id AND
```

Comment:
I feel like we are missing some condition. This pairs canonical profiles with all in use profiles for a current account just by `ref_id` and `os_major_version`. This compares only profiles and not benchmarks which parents. I'm wondering if we should add or replace `os_major_version` with benchmark `ref_id`.

Reply:
`os_major_version` IS (derived from) benchmark `ref_id`

Reply:
Yes, it is, today. :-)

Reply:
We're in a big trouble if that changes :see_no_evil:

Reply:
Logically it would be better to match on benchmarks as well. Would that increase the SQL load? I believe that the `os_major_version` comes from benchmark nevertheless.

Reply:
I had benchmark `ref_id` in mind which would be similar to OS major check, but seemingly logically more correct (?).

Reply:
And with 3x the complexity :sweat:

Reply:
Check the `OS_MAJOR_VERSION` field definition :smile:, that is more complex compared to ref_id.